### ISSUE TITLE
Advanced User toggle

### ISSUE THREAD

**Issue by scruffynerf** [stashapp/stash#4202]
I'd really like to see an "advanced user" toggle.  And then based on that toggle, we hide all manner of confusing switches/tools/etc from the UI.
That lets us leave the items intact, but avoid people using who don't understand them.

For example the transcoding discussion and while a new tool tip more helpful, it's a gotcha if you use it when you don't MEAN to pre-transcode.  Only "slow cpus" should ever turn this on.  So make it advanced, so we can just say "have you tried turning this on?", Instead of "oops ,you turned that on".  Same for animated preview, for Auto Tag, etc.

Just cover the sharp edges and electric outlets for the newbies.
So essentially all UI just needs to respect the Advanced User toggle and hide the items.  Heck, just make them hidden via css...

WP requested a list of items:

> I don't think this would be something too difficult for someone to do. If you can raise this an an issue, with a list of what should hidden under Advanced settings, then I can put it in the milestone for this release.
> I'd probably put the toggle in the sidebar, and ensure it's persisted in the stored ui settings.

We can compile the list below...

**Comment by scruffynerf** [comment]
first pass:

Tasks:
- Scan:
  - Generate animated image previews
- Auto Tag
- Generate:
  - Animated Image Previews
  - Marker Animated Image Previews
  - Transcodes

- Migrations ?? (at this point, new users shouldn't need them)

Interface:
- Scene / Marker Wall
  - Preview Type   - add comment to animated to enable advantage?, or require advanced toggle is on to pick?
   
Entire Services tab

System:
- Hashing

**Comment by QxxxGit** [comment]
Perhaps this toggle should be implemented into the installation process, with a link to documentation that displays the list of hidden items?

My list below is based upon what I *think* most basic users won't need to use/interact with.

* Tasks
    * Anonymise section
    * Migrations
* Interface
    * Custom Localization section
* System
    * Hashing
    * Transcoding

**Comment by QxxxGit** [comment]
> first pass:
> 
> Tasks:
> * Auto Tag

Curious about your thoughts on this @scruffynerf, as it seems important to link everything up. When I was introducing a user to Stash, their priorities were to scan the content in, and how to find it easily. I instructed them to create performers, maybe some tags, and how auto tagging works to link it up. Similar to me, their use-case was more tailored for YouTube, OnlyFans, etc. content so StashDB wouldn't be of much use.

From my understanding, you're much more involved with new users so I would definitely defer to you. Just curious if this should be taken into consideration.

**Comment by scruffynerf** [comment]
> Curious about your thoughts on this @scruffynerf, as it seems important to link everything up. When I was introducing a user to Stash, their priorities were to scan the content in, and how to find it easily. I instructed them to create performers, maybe some tags, and how auto tagging works to link it up. Similar to me, their use-case was more tailored for YouTube, OnlyFans, etc. content so StashDB wouldn't be of much use.

You're the exception then... FansDB is slowly growing, and will do OF, etc.
The average user sees 'auto tag', says That's what I want.. and then complains it didn't do the 'thing'

Auto Tag depends on good filenames, with Studios, Stars, Titles.. in other words, the content is already organized.
Why would that be assumed?  If you have organization, and you want to migrate to Stash, that's still not the best method, we still want you to use StashDB/etc first, because Auto Tag at best does a crappy job of adding just 3 fields.

By your logic, we should promote FIle Parser over Auto Tag anyway.  Have you use File Parser at all?

**Comment by DogmaDragon** [comment]
I think we should ask new people what they consider advanced features. Veteran users have skewed perspective.

**Comment by scruffynerf** [comment]
Agreed. And was already asking a few newbies to comment here.  This list is meant as a starting point.

**Comment by xantror** [comment]
Curious how you want to handle this change in the in-app manual. It might need to be overhauled to reflect some options being hidden.

**Comment by QxxxGit** [comment]
> You're the exception then... FansDB is slowly growing, and will do OF, etc. The average user sees 'auto tag', says That's what I want.. and then complains it didn't do the 'thing'
> 
> Auto Tag depends on good filenames, with Studios, Stars, Titles.. in other words, the content is already organized. Why would that be assumed? If you have organization, and you want to migrate to Stash, that's still not the best method, we still want you to use StashDB/etc first, because Auto Tag at best does a crappy job of adding just 3 fields.
> 
> By your logic, we should promote File Parser over Auto Tag anyway. Have you used File Parser at all?

Outside of Stash, I've seen folks with a similar use-case to mine typically store files within directories based on performer name, and sometimes subdirectories that focus on body features. To me, that's not far-fetched to assume that degree of organization. They're not using a complex file structure that includes studios or dates, or hundreds/thousands of performers, because they're using the file explorer to quickly get to their content of like a dozen performers.

Linking records through Auto Tag, I think, seems to be a very basic and necessary feature for this use-case. And what about the odd-out images? If I'm a basic user, how do I link those up to my performers and tags if Auto Tag is hidden?

**Comment by scruffynerf** [comment]
I think you missed my point: Auto Tag does a crappy job of doing the right thing.  It's not the best tool.  It's a limited use, half ass job, so we should not encourage it's use.  If you are encouraging it, you are doing the new user a disservice.  They end up with BARELY metadata-ed content.

**Comment by Dark-Obsidian** [comment]
Hi guys, this is something I have also been considering suggesting over the past week or so. Have added my thoughts below...

### RE: Advanced User toggle: UI control
- I would suggest rather than a literal toggle, i.e. a binary option (presumably implemented as a checkbox)
- Implement this as a dropdown option where the user can select a "settings level" from a list of options, such as: `Basic`, `Standard`, `Advanced`, `Expert`
- This approach allows for greater flexibility and future extensibility (e.g. this way it can be still be implemented initially as a binary choice if that's preferred (e.g. `Standard` or `Advanced`) but also allows for a potential 3rd additional level (e.g. `Expert`) to be added later without any refactoring
- In addition, this allows for a smoother transition between options presented -- i.e. rather than toggling between showing ***some options* and ***everything***, it allows for this to staggered across levels...
- **EXAMPLE** `Settings > System`...
  - `Standard`: Shows common options like: `Application Paths` and `Logging`, but no transcoding options
  - `Advanced`: Shows above, plus some transcoding options, like: `Hardware encoding` and Maximum transcode size`
   - `Expert`: Shows above, plus all transcoding options, including: `FFmpeg Live Transcode Input Args` for example

***

### RE: Advanced User toggle: Specific mappings
Referring to the exact settings that would be mapped to the control, e.g. Scruffy's post above, showing...
```
Tasks
 └  Scan
      └  Generate animated image previews
 └  Auto-Tag
 └  Generate
      └  Animated Image Previews
      └  Marker Animated Image Previews
      └  Transcodes
```
... Similarly have some thoughts on this as have been considering suggesting some edits to the `settings > tasks` page myself, however:
- (a) I don't want to bog down the ***implementation of the control*** with an extended debate about exactly how each option would be linked to it
- (b) If the suggestion above of having this as a [dropdown control rather than a toggle] is agreed, propose option mapping is picked up by separate group / chat -- e.g. with a CSV export of every sub-setting and a column showing which level it would be mapped to

***

### RE: Auto-Tag / Identity
Before people reply with "auto-tag is not the same as identify, they are different things", I have included this as they are at least connected and I have had both positive experiences and quite negative/concerning experiences with this area of stash. To elaborate...

#### ✅ Positive experiences...
- I have had a pretty good experience using the "specific" `Auto-Tag` option, i.e. the Auto-Tag button on an individual performer's / studio's / tag's page
- Typically, I will tag several scenes in `Scene Tagger` and then switch to the page of the newly-tagged performer, etc and click the `Auto-Tag` button...
- Whilst this can seem a *little* tedious (if I have just added say 20x performers), overall I have found this to be a great way of helping to get my stash database populated as quickly as possible
- Anecdotally, it also "feels good" to add a new performer with `Scene Tagger` (so they have a picture, stats, etc) press the `Auto-Tag` button, refresh the page and see that I have gone from having this performer tagged in 1 scene, to say 30 👍🏼

#### ⚠️ Negative / concerning experiences...
- Despite the above, tagging scenes has felt very laborious and slow at times [see: **PLACEHOLDER** `RFC: Scene Tagger not matching scenes it presumably should`†] and so I decided to risk trying the `Identify` option...
- To address the point of why I am including `Identify` in this -- `Identify` includes `Auto Tag` as one of the scrapers it runs [I assume this `Auto Tag` is literally the same as the clicking the `Auto Tag` option in settings]
- Whilst this job maybe helped get my DB populated (I noticed a big increase in the number of performers for instance), any improvements or positive perceptions were immediately over-shadowed by...
   - A number of *technically correct* but ***unwanted entries*** items added [see: **PLACEHOLDER** `RFC: Dealing with pseudo studios and sub-brands`†]
   - Several ***definitely wrong*** (and frankly inexplicable) scenes being matched [see: **PLACEHOLDER** `RFC: Identify making some inexplicable and bizarre scene matches`†], which had the knock-on impact of adding junk studios and performers - which all had to be un-done afterwards.

> † I will write this up today/tomorrow and replace the link once posted.

While the ***overall impact*** of using `Identify` (and indirectly `Auto Tag`) may have been positive, i.e. statistically it probably added more correct scenes than wrong ones -- the ***perceived impact*** was arguably negative!
- After running `Identify` (with `Auto Tag`) - ***I had to spend over an hour*** un-doing large parts of what it did and deleting numerous: wrong performers, duplicate studios, unwanted studios, mis-matched scenes, etc from my stash
- As part of some of the documentation updates I'd like to submit over the next few weeks, I want to be clear with new users that getting all your content imported and organised in stash will ***not*** be as quick / easy as using say Kodi...
   - I think making this clearer upfront will help avoid user fatigue and frustration later on
   - I believe the "need to put several hours aside to go through the process of importing your content" can be sold by pointing out things like the *MUCH* bigger set of possible matches, generally less-clear filenames, etc
   - ***However*** the above is severely undermined if users feel tools like `Auto Tag` and/or `Identify` are "booby traps" or that they "screw up the database" and "can't be trusted"

***

That ended up being a bigger post that I intended...
### TLDR
- Agree with some kind of "Advanced settings" option, but think a dropdown control will work better than a toggle/binary option
- I am keen to input into the "settings to level-mapping" conversation but suggest de-coupling it from this feature request to avoid bogging it down
- `Auto Tag` has worked well when used on a specific performer / studio page, but has also done a lot of damage when the global option has been used from settings / as part of `Identify`

**Comment by scruffynerf** [comment]
from a recent comment in Discord help:

> (Identify is) the first thing that you see when you are new to stash. The tagger button (even though described pretty well in the on-boarding passage) is somewhat hidden and less obvious. It is named 'tagger', well as I was new, I wanted to identify my scenes and get an overview over my collection, not tag scenes. 

The order in which we present items, or fail to present items to new users leads to them using or not using them.
That's the entire point of this:  How do we keep the existing items that _new_ users shouldn't be using, but hide them?
How do we de-emphasize someone thinking Identify is what they want (and then they screw up the settings, which gives them nothing they wanted, and makes it worse), when Tagger would have been very easy and clickable.

Catering to the 'I don't care, I just want it mostly right, and to do no work' crowd is NOT a good thing in the long run, and causes support issues, who blame Stash for them not being what they wanted out of it.

**Comment by xantror** [comment]
Maybe a lower impact option can be implemented (a middle ground):

Instead of putting items behind a global toggle, make them less visible and mark them as "advanced" by putting them under a dropdown

![HyUvVnPafG](https://github.com/stashapp/stash/assets/69341873/df200779-861d-4d58-a0ab-ae0e7d6884c6)

**Comment by scruffynerf** [comment]
> Instead of putting items behind a global toggle, make them less visible and mark them as "advanced" by putting them under a dropdown

doable but potentially way more UI work.

**Comment by Dark-Obsidian** [comment]
I appreciate my last post earlier talked about `Identify` and `Auto Tag` quite a bit. This wasn't my intention when I began writing but included it as I had recently experienced these for the first time and thought that this context may have been helpful, however I don't want to distract from the main topic...

To re-characterise what I wrote above...
- I think @scruffynerf's feature request for a control that can be used to hide more advanced settings, unless the user explicitly selects to see this is a **great idea** and 100% agree with this being added to stash
- FYI: This is something that other content organiser apps like Kodi and Plex already include
- Personally, I think a "settings level ***dropdown***" would offer more flexibility than a "**toggle**" (checkbox)
- I think the ability to hide more advanced controls goes beyond specific functions like `Identify` and `Auto Tag` -- again from my experience as a new user, I was quite confused by what things like `heatmaps` were in this context and `funscripts`, etc... so would suggest not binding the **definite benefits** of this control, which too much specific debate at this stage over exactly which controls get hidden, etc -- that can be decided later.

PS -- I do like @xantror's UI mockup -- thanks for taking the time to create and share that! FYI: this is how `Plex` implement it... however I also agree that this may involve more UI work, and may create complication of either items becoming out of order (if done at a "page level") or needing to have multiple "advanced settings" buttons per page (if done at a "section level")

**Comment by Dark-Obsidian** [comment]
> @elkorol: I think if this is getting implemented it would be beneficial to implement a settings search also like we have with the edit filters modal dialog maybe including but not necessary the ability to pin certain settings to the top.

Not a bad idea... but think maybe settings search / option to pin various settings to top should be a separate(?)

**Comment by lesderid** [comment]
> * Several _**definitely wrong**_ (and frankly inexplicable) scenes being matched [see: **PLACEHOLDER** `RFC: Identify making some inexplicable and bizarre scene matches`†], which had the knock-on impact of adding junk studios and performers - which all had to be un-done afterwards.

I've seen this happen quite a few times, too.
